addons/mail/models/discuss/discuss_channel.py

```diff
@@ -505,10 +508,12 @@ def add_members(
                     payload["invited_by_user_id"] = self.env.user.id
                 member._bus_send("discuss.channel/joined", payload)
                 if post_joined_message:
+                    message_author = current_partner if current_partner else current_guest
+                    member_name, author_name = self._get_user_names(member=member, author=message_author)
```

Comment:
function `_get_user_names` looks too specific, of having a member and author parameter.
Would feel more natural as `member.get_displayed_name()`.